verde/utils.py

```diff
@@ -254,6 +255,73 @@ def grid_to_table(grid):
     return data
 
 
+def load_surfer(fname, dtype="float64"):
+    """
+    Read data from a Surfer ASCII grid file.
+
+    Surfer is a contouring, griding and surface mapping software
+    from GoldenSoftware. The names and logos for Surfer and Golden
+    Software are registered trademarks of Golden Software, Inc.
+
+    http://www.goldensoftware.com/products/surfer
+
+    This function opens a Surfer grid file, masks any NaNs in the data,
+    and outputs a :class:`xarray.DataArray` that contains easting and northing
+    coordinates, data values, and associated file metadata in the
+    DataArray attributes.
+
+    Parameters
+    ----------
+    fname : str
+        Name or path of the Surfer grid file
+    dtype : str or numpy dtype
+        The type of variable used for the data. Default is ``float64``. Use
+        ``float32`` if the data are large and precision is not an issue.
+    Returns
+    ----------
+    data : :class:`xarray.DataArray`
+        A 2D grid with the data variables.
+    """
+    # Surfer ASCII grid structure
+    # DSAA            Surfer ASCII GRD ID
+    # nCols nRows     number of columns and rows
+    # xMin xMax       X min max
+    # yMin yMax       Y min max
+    # zMin zMax       Z min max
+    # z11 z21 z31 ... List of Z values
+    with open(fname) as input_file:
+        # DSAA is a Surfer ASCII GRD ID
+        grd_id = input_file.readline().strip()
+        # Read the number of columns (ny) and rows (nx)
+        ydims, xdims = [int(s) for s in input_file.readline().split()]
+        # Our x points North, so the first thing we read is y, not x.
+        south, north = [float(s) for s in input_file.readline().split()]
+        west, east = [float(s) for s in input_file.readline().split()]
+        dmin, dmax = [float(s) for s in input_file.readline().split()]
+        field = np.fromiter(
+            (float(s) for line in input_file for s in line.split()), dtype=dtype
+        )
+        nans = field >= 1.70141e38
+        if np.any(nans):
+            field = np.ma.masked_where(nans, field)
+        err_msg = "{} of data ({}) doesn't match one from file ({})."
+        if not np.isclose(len(field), ydims * xdims):
```

Comment:
```suggestion
        if field.size != ydims * xdims:
```

The `np.isclose` function is used to test floating point numbers. Since all are integers `!=` will work better. With a numpy array, it's best to check `.size` or `.shape` since `len` of a 2D array will be number of rows instead of the number of elements. I've had some bugs creep up in the past because of that.

Reply:
Right on, thank you for the feedback. I have commited the changes, and will update the tests that I am working on